src/Impostor.Server/Events/Game/BeforeGameCreatedEvent.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Impostor.Api.Events;
+using Impostor.Api.Games;
+using Impostor.Api.Innersloth;
+using Impostor.Server.Net.Redirector;
+using Impostor.Server.Net.State;
+
+namespace Impostor.Server.Events
+{
+    public class BeforeGameCreatedEvent : IBeforeGameCreatedEvent
+    {
+        private readonly ICollection<int> _gameKeys;
+
+        public BeforeGameCreatedEvent(ICollection<int> gameKeys)
+        {
+            _gameKeys = gameKeys;
+        }
+
+        public GameCode? GameCode { get; private set; }
+
+        public bool TryToSetCode(GameCode gameCode)
+        {
+            if (gameCode.Code.Length != 6)
```

Comment:
What if `GameCode` is v1 (or better, v3?). And code for it is already present in [`GameCodeParser`](https://github.com/Impostor/Impostor/blob/dc759b1d573b59ff5c39dd2792c40ebdec7649a2/src/Impostor.Api/Innersloth/GameCodeParser.cs#L95) anyways, just check if value is -1, or better, make a new property in GameCode which is just `IsInvalid => Value == -1`